File: screenshot.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const distDir = path.resolve(__dirname);
const screenshotDir = path.join(distDir, 'screenshots');
if (!fs.existsSync(screenshotDir)) fs.mkdirSync(screenshotDir);

const pages = [
  { file: 'index.html', slug: 'index' },
  { file: 'junzaiwang.html', slug: 'junzaiwang' },
  { file: 'kelionwang.html', slug: 'kelionwang' },
  { file: 'duoluanzhuang.html', slug: 'duoluanzhuang' },
  { file: 'zhangbiaowang.html', slug: 'zhangbiaowang' },
  { file: 'zujinwang.html', slug: 'zujinwang' },
  { file: 'baomiaowang.html', slug: 'baomiaowang' },
];

const viewports = [
  { name: 'desktop', width: 1280, height: 900 },
  { name: 'mobile', width: 390, height: 844 },
];

(async () => {
  const browser = await chromium.launch({ headless: true });
  let count = 0;

  for (const vp of viewports) {
    const ctx = await browser.newContext({ viewport: { width: vp.width, height: vp.height } });
    const page = await ctx.newPage();

    for (const pg of pages) {
      const url = 'file:///' + path.join(distDir, pg.file).replace(/\\/g, '/');
      await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
      // wait for fonts (brief)
      await page.waitForTimeout(800);
      const outFile = path.join(screenshotDir, `${pg.slug}_${vp.name}.png`);
      await page.screenshot({ path: outFile, fullPage: false });
      console.log(`[OK] ${pg.slug}_${vp.name}.png`);
      count++;
    }

    await ctx.close();
  }

  await browser.close();
  console.log(`\nDone: ${count} screenshots in ${screenshotDir}`);
})();

File: screenshot.py
import os, sys
from pathlib import Path
from playwright.sync_api import sync_playwright

dist_dir = Path(__file__).parent.resolve()
screenshot_dir = dist_dir / "screenshots"
screenshot_dir.mkdir(exist_ok=True)

pages = [
    ("index.html", "index"),
    ("junzaiwang.html", "junzaiwang"),
    ("kelionwang.html", "kelionwang"),
    ("duoluanzhuang.html", "duoluanzhuang"),
    ("zhangbiaowang.html", "zhangbiaowang"),
    ("zujinwang.html", "zujinwang"),
    ("baomiaowang.html", "baomiaowang"),
]

viewports = [
    ("desktop", 1280, 900),
    ("mobile", 390, 844),
]

count = 0
with sync_playwright() as pw:
    browser = pw.chromium.launch(headless=True)
    for vp_name, width, height in viewports:
        ctx = browser.new_context(viewport={"width": width, "height": height})
        page = ctx.new_page()
        for file_name, slug in pages:
            url = "file:///" + str(dist_dir / file_name).replace("\\", "/")
            page.goto(url, wait_until="networkidle", timeout=30000)
            page.wait_for_timeout(800)
            out = screenshot_dir / f"{slug}_{vp_name}.png"
            page.screenshot(path=str(out), full_page=False)
            print(f"[OK] {slug}_{vp_name}.png")
            count += 1
        ctx.close()
    browser.close()

print(f"\nDone: {count} screenshots -> {screenshot_dir}")
